### ISSUE TITLE
Change popup initialization to use polling for most accuracy in triggering

### ISSUE THREAD

**Issue by danieliser** [PopupMaker/Popup-Maker#439]
Currently popups are initialized all at once on page load. Triggers are activated and when they finally go to open the popup conditions are tested. This means that a time delay trigger won't work if conditions fail, even if conditions on that page view change to allow it.

Polling method:

Every x delay (likely 1 second)  test each popups conditions, if they pass activate the popup.

On activation start triggers which may still retest conditions at the time of opening to ensure they still pass.

Polling may need 2 runs.

- Testing inactive popups to see if they should be activated.
- Testing active popups to see if they should be deactivated.

This will require making triggers attachable/detachable like I already did in Ahoy.

This also may be best done with a rebuild of front end rendering via JS templates for an extremely clean DOM.

**Comment by Bessonweb** [comment]
Hi, have you some news about this issue?

**Comment by danieliser** [comment]
@Bessonweb - Not yet but mainly because I am spread to thin currently. I am hiring a full time developer (or trying to) at the moment, which I hope to have working on nothing but Popup Maker going forward.

Currently I'm managing multiple plugins, all their support, marketing & business side of things by myself. Decided this isn't moving things quick enough so things are changing.

**Comment by Bessonweb** [comment]
@danieliser I understand your position. I did not know you was only one dev behind Popup Maker!

I would like to propose you my help (not in dev) but I'm overwhelmed too.

I wish you a good continuation!

**Comment by danieliser** [comment]
@Bessonweb - Same to you.

**Comment by danieliser** [comment]
@fpcorso Should this be part of the bigger front end JS rewrite? I want to do some restructuring anyways.

Probably more functional based like Ahoy instead of using the custom jQuery plugin we have now.

Of course maintain backward compatibility layer.

**Comment by fpcorso** [comment]
@danieliser , yeah, if we are doing a bigger JS re-write anyways, maybe we should map those out into a major epic and put all these together. Would help prevent us from having to re-do any of our changes as we go.

**Comment by danieliser** [comment]
@fpcorso Yea each of these is a seem small, but so many things are tied into initialization. Triggers for example need to be rewritten to support activating/deactivating.

There is currently no mechanism to remove an active trigger. This exists in Ahoy, we will be using that as a model of sorts.
